kotlin/services/ec2/src/main/kotlin/com/kotlin/ec2/EC2Scenario.kt

```diff
@@ -0,0 +1,528 @@
+        ec2.waitUntilInstanceStopped { // suspend call
+            instanceIds = listOf(instanceId)
+        }
+        println("Successfully stopped instance $instanceId")
+    }
+}
+// snippet-end:[ec2.kotlin.scenario.stop_instance.main]
+
+suspend fun describeEC2InstancesSc(newInstanceId: String?): String {
+    var pubAddress = ""
+    var isRunning = false
+    val request = DescribeInstancesRequest {
+        instanceIds = listOf(newInstanceId.toString())
+    }
+
+    while (!isRunning) {
+        Ec2Client { region = "us-west-2" }.use { ec2 ->
+            val response = ec2.describeInstances(request)
+            val state = response.reservations?.get(0)?.instances?.get(0)?.state?.name?. value
+            if (state != null) {
+                if (state.compareTo("running") == 0) {
+                    println("Image id is ${response.reservations!!.get(0).instances?.get(0)?.imageId}")
+                    println("Instance type is ${response.reservations!!.get(0).instances?.get(0)?.instanceType}")
+                    println("Instance state is ${response.reservations!!.get(0).instances?.get(0)?.state}")
+                    pubAddress = response.reservations!!.get(0).instances?.get(0)?.publicIpAddress.toString()
+                    println("Instance address is $pubAddress")
+                    isRunning = true
+                }
+            }
+        }
+    }
+    return pubAddress
+}
+
+suspend fun runInstanceSc(instanceTypeVal: String, keyNameVal: String, groupNameVal: String, amiIdVal: String): String {
+    val runRequest = RunInstancesRequest {
+        instanceType = InstanceType.fromValue(instanceTypeVal)
+        keyName = keyNameVal
+        securityGroups = listOf(groupNameVal)
+        maxCount = 1
+        minCount = 1
+        imageId = amiIdVal
+    }
+
+    Ec2Client { region = "us-west-2" }.use { ec2 ->
+        val response = ec2.runInstances(runRequest)
+        val instanceId = response.instances?.get(0)?.instanceId
+        println("Successfully started EC2 Instance $instanceId based on AMI $amiIdVal")
+        return instanceId.toString()
+    }
+}
+
+// snippet-start:[ec2.kotlin.describe_instance_type.main]
+// Get a list of instance types.
+suspend fun getInstanceTypesSc(): String {
+    var instanceType = ""
+    val filterObs = ArrayList<Filter>()
+    val filter = Filter {
+        name = "processor-info.supported-architecture"
+        values = listOf("arm64")
+    }
+
+    filterObs.add(filter)
+    val typesRequest = DescribeInstanceTypesRequest {
+        filters = filterObs
+        maxResults = 10
+    }
+    Ec2Client { region = "us-west-2" }.use { ec2 ->
+        val response = ec2.describeInstanceTypes(typesRequest)
+        response.instanceTypes?.forEach { type ->
+            println("The memory information of this type is ${type.memoryInfo?.sizeInMib}")
+            println("Network information is ${type.networkInfo}")

```

Comment:
I think it is printing the type here, you may want to look at it and print some specific properties in a more readable format. Here is what I see: 
NetworkInfo(defaultNetworkCardIndex=0,efaInfo=null,efaSupported=false,enaSupport=required,encryptionInTransitSupported=false,ipv4AddressesPerInterface=50,ipv6AddressesPerInterface=50,ipv6Supported=true,maximumNetworkCards=1,maximumNetworkInterfaces=15,networkCards=[NetworkCardInfo(maximumNetworkInterfaces=15,networkCardIndex=0,networkPerformance=25 Gigabit)],networkPerformance=25 Gigabit)